pylint/checkers/variables.py

```diff
@@ -2596,23 +2600,24 @@ def _loopvar_name(self, node: astroid.Name) -> None:
                     ):
                         return
 
-        maybe_walrus = utils.get_node_first_ancestor_of_type(node, nodes.NamedExpr)
-        if maybe_walrus:
-            maybe_comprehension = utils.get_node_first_ancestor_of_type(
-                maybe_walrus, nodes.Comprehension
+        if (
+            walrus := utils.get_node_first_ancestor_of_type(node, nodes.NamedExpr)
+        ) and (
+            comprehension := utils.get_node_first_ancestor_of_type(
+                walrus, nodes.Comprehension
             )
-            if maybe_comprehension:
-                comprehension_scope = utils.get_node_first_ancestor_of_type(
-                    maybe_comprehension, nodes.ComprehensionScope
-                )
-                if comprehension_scope is None:
-                    # Should not be possible.
-                    pass
-                elif (
-                    comprehension_scope.parent.scope() is scope
-                    and node.name in comprehension_scope.locals
-                ):
-                    return
+        ):
+            comprehension_scope = utils.get_node_first_ancestor_of_type(
+                comprehension, nodes.ComprehensionScope
+            )
+            if comprehension_scope is None:
+                # Should not be possible.
+                pass
+            elif (
+                comprehension_scope.parent.scope() is scope
+                and node.name in comprehension_scope.locals
+            ):
+                return
```

Comment:
Tbh I'm not sure this is really an improvement if it's just to work around the `too-many-statements` error. We already disable `too-many-branches` for this function, why not `too-many-statements` as well.

https://github.com/pylint-dev/pylint/blob/51c28bc922dfb8da42e680ed7190eb942889df01/pylint/checkers/variables.py#L2524-L2525

```diff
-    # pylint: disable = too-many-branches
+    # pylint: disable-next=too-many-branches,too-many-statements
```